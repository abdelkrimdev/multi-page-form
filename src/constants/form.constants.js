export const formConstants = {
  NEXT_STEP: 'GOTO_NEXT_STEP',
  PREV_STEP: 'GOTO_PREVIOUS_STEP',
  UPDATE_FORM: 'UPDATE_FORM_DATA'
}
