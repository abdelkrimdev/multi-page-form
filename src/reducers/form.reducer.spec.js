import { formConstants } from '../constants/form.constants'
import { form } from './form.reducer'

describe('form reducer', () => {
  it('should have initial state.', () => {
    const action = { type: 'DUMMY_ACTION' }

    expect(form(undefined, action)).toMatchSnapshot()
  })

  it('should return the current state after receiving a dummy action.', () => {
    const state = { data: 'current state' }
    const action = { type: 'DUMMY_ACTION' }

    expect(form(state, action)).toMatchSnapshot()
  })

  it('should handle start registration action.', () => {
    const action = { type: formConstants.START_REGISTRATION }

    expect(form({ }, action).step).toEqual(1)
  })

  it('should handle next step action.', () => {
    const state = { step: 1 }
    const action = {
      type: formConstants.NEXT_STEP
    }

    expect(form(state, action).step).toEqual(state.step + 1)
  })

  it('should handle previous step action.', () => {
    const state = { step: 2 }
    const action = {
      type: formConstants.PREV_STEP
    }

    expect(form(state, action).step).toEqual(state.step - 1)
  })

  it('should handle update form action.', () => {
    const state = {
      step: 2,
      name: 'Clark'
    }
    const action = {
      type: formConstants.UPDATE_FORM,
      payload: {
        key: 'address',
        value: 'Krypton'
      }
    }

    expect(form(state, action)).toMatchSnapshot()
  })
})
