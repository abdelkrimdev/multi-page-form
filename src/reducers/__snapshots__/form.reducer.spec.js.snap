// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form reducer should handle update form action. 1`] = `
Object {
  "address": "Krypton",
  "name": "Clark",
  "step": 2,
}
`;

exports[`form reducer should have initial state. 1`] = `Object {}`;

exports[`form reducer should return the current state after receiving a dummy action. 1`] = `
Object {
  "data": "current state",
}
`;
